tests/test_flows.py

```diff
@@ -695,17 +717,63 @@ def test_ap_capability_query(self):
         time.sleep(1)
 
-        debug("Confirming link metric query has been received on agent")
-        self.check_log(env.agents[0], "Received LINK_METRIC_QUERY_MESSAGE")
-
-        debug("Confirming link metric response has been received on controller")
-        self.check_log(env.controller, "Received LINK_METRIC_RESPONSE_MESSAGE")
-        self.check_log(env.controller, "Received TLV_TRANSMITTER_LINK_METRIC")
-        self.check_log(env.controller, "Received TLV_RECEIVER_LINK_METRIC")
+        query = self.check_cmdu_type_single("link metric query", 0x0005,
+                                            env.controller.mac, env.agents[0].mac,
+                                            mid)
+        query_tlv = self.check_cmdu_has_tlv_single(query, 0x08)
+
+        debug("Checking query type and queried metrics are correct")
+        self.safe_check_obj_attribute(query_tlv, 'link_metric_query_type', str(0x00),
+                                      "Query type is not 'All neighbors'")
+        self.safe_check_obj_attribute(query_tlv, 'link_metrics_requested', str(0x02),
+                                      "Metrics for both Tx and Rx is not requested")
+
+        resp = self.check_cmdu_type_single("link metric response", 0x0006,
+                                           env.agents[0].mac, env.controller.mac,
+                                           mid)
+
+        debug("Checking response contains expected links to/from agent and nothing else")
+        # neighbor macs are based on the setup in "launch_environment_docker" method
+        expected_tx_link_neighbors = [env.controller.mac, env.agents[1].mac]
+        expected_rx_link_neighbors = [env.controller.mac, env.agents[1].mac]
+        actual_tx_links = self.check_cmdu_has_tlvs(resp, 0x09)
+        actual_rx_links = self.check_cmdu_has_tlvs(resp, 0x0a)
+
+        def verify_links(links, expected_neighbors, link_type):
+            verified_neighbors = []
+            unexpected_neighbors = []
+            for link in links:
+                self.safe_check_obj_attribute(link, 'responder_mac_addr', env.agents[0].mac,
+                                              "Responder MAC address is wrong")
+
+                try:
+                    # a tshark (v3.2.4) bug causes "neighbor_mac_addr" field to
+                    # show up as "responder_mac_addr_2"
+                    if link.responder_mac_addr_2 in expected_neighbors:
+                        verified_neighbors += [link.responder_mac_addr_2]
+                    else:
+                        unexpected_neighbors += [link.responder_mac_addr_2]
+                except AttributeError as e:
+                    self.fail(e)
+
+            # we expect each neighbor to appear only once
+            for neighbor, count in Counter(verified_neighbors).items():
+                if count != 1:
+                    self.fail("{} link to neighbor {} appears {} times"
```

Comment:
Worth to add expected amount of appears in `self.fail` message.
```suggestion
                    self.fail("{} link to neighbor {} appears {} times. We expect it to appear only once."
```

Reply:
I wanted to but I was not sure how to say it :+1: